rootfs/etc/nginx/lua/test/defer/defer_test.lua

```diff
@@ -0,0 +1,18 @@
+package.path = "./rootfs/etc/nginx/lua/?.lua;./rootfs/etc/nginx/lua/test/mocks/?.lua;" .. package.path
+_G._TEST = true
+local defer = require('defer')
+
+local _ngx = {
+    shared = {},
+    log = function(...) end,
+    get_phase = function() return "timer" end,
+}
+  _G.ngx = _ngx
+  
+describe("Defer", function()
+   describe("to_timer_phase", function()
+    it("executes passed callback immediately if called on timer phase", function()
+        assert.equal(true, defer.to_timer_phase(function() return 1 end))
```

Comment:
how does this assert that the function was executed?

Reply:
👍 Create a table here - and have the callback modify the table's value. E.g. Initialize `defer.call_count = 0`, and then in the callback - increment that value by 1. 

Your assertion can check for that value having been increased. 

The problem with this right now is that your test will pass regardless of whether or not the callback passed in to `to_timer_phase` is executed.